### ISSUE TITLE
Sprint: Aug 29 → Sept 4

### ISSUE THREAD

**Issue by RichardLitt** [ipfs/pm#156]
# Sprint August 29th

_For information about these calls and how they work, read [the ipfs/pm README](https://github.com/ipfs/pm)._

[Calendar](https://calendar.google.com/calendar/embed?src=ipfs.io_eal36ugu5e75s207gfjcu0ae84@group.calendar.google.com)

Endeavour         | Lead            | Time (PDT - **UTC/Z** - CEST) | Pad
:---------------: | :-------------: | :--------------------------: | :----:
All Hands Call   | @flyingzumwalt    | 9:00PDT **16:00Z** 18:00CEST  | [all-hands notes](https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-all-hands)
orbit               | @haadcode   | 9:30PDT **16:30Z** 16:30CEST | [orbit notes](https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-orbit)
go-ipfs               | @whyrusleeping  | 10:00PDT **17:00Z** 19:00CEST | [go-ipfs notes](https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-go-ipfs)
js-ipfs                | @diasdavid       | 10:30PDT **17:30Z** 19:30CEST | [js-ipfs notes](https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-js-ipfs)
libp2p            | @diasdavid      | 11:00PDT **18:00Z** 20:00CEST | [libp2p notes](https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-libp2p)

Please add any agenda items you have to the pad before the project-specific sprint call starts.

**Comment by RichardLitt** [comment]
I added a Mediachain call; @parkan let me know if this sounds good to you.

**Comment by flyingzumwalt** [comment]
Agenda item: A waffle board for ipfs/pm (and maybe add ipfs/community) https://waffle.io/ipfs/pm

**Comment by dignifiedquire** [comment]
@RichardLitt can you please update the calendar?

**Comment by diasdavid** [comment]
I've added agenda items to js-ipfs and libp2p, please do so too if you have any.

- https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-js-ipfs
- https://public.etherpad-mozilla.org/p/ipfs-2016-08-29-libp2p

**Comment by parkan** [comment]
@RichardLitt one of our devs (who I would really like to participate) is in EEST (Greece) so this is a bit late for him, is it possible to do it earlier in the day tmw?

**Comment by RichardLitt** [comment]
@parkan Sorry, I should have gotten to this sooner (still figuring out the new system). Let's schedule a call for Wednesday?

**Comment by parkan** [comment]
Sounds good to me -- and I should have responded earlier too, but was on a long weekend 😄

**Comment by RichardLitt** [comment]
No worries! My bad for not catching this. :)

**Comment by lgierth** [comment]
## IPFS All Hands Call August 29th

**Moderator:**  @dignifiedquire
**Notetaker:**  @lgierth

### Attendees

  * @RichardLitt 
  * @diasdavid
  * @victorbjelkholm
  * @Kubuxu
  * @flyingzumwalt
  * @haadcode
  * @jbenet
  * @em-ly
  * @dignifiedquire

### Agenda

1. Roll call by alphabetical order (moderator)
2. Call for additional agenda items (moderator)
3. Milestones and (experimentally) a waffle board for ipfs/pm (and maybe include ipfs/community) https://waffle.io/ipfs/pm (@flyingzumwalt)
  - milestone 1: Implement new PM Model https://github.com/ipfs/pm/milestone/1
  - milestone 2: First Org Roadmap https://github.com/ipfs/pm/milestone/2
4. go-ipfs v0.4.3 review + changes
5. Newsletter
6. Identity Providers (Orbit and more) -- call?
7. Community announcement: Code for Philly distributed systems "course" https://github.com/ipfs/community/issues/162
8. Moderator/notetaker for next time:
  - Moderator: @lgierth
  - Notetaker: @em-ly

### Notes

- **!!!we hit the hangouts participants limit!!! (again)**
- milestones FYI, em-ly and flyingzumwalt will follow up
- The go-ipfs CR list: https://github.com/ipfs/go-ipfs/issues/3138 see the MUST part
  - will follow up in the go-ipfs hangout
- newsletter
  - supposed to cover everything since the workshop in lisbon
  - review it and add stuff, everone!
  - @richardlitt wants to ship it shortly
  - https://github.com/ipfs/newsletter/pull/52
- identity providers
  - @haadcode's integrating orbit and uport
  - there's more possible identity providers
  - e.g. the estonian e-citizenship
- @flyingzumwalt working on distributed systems course
  - walking from "i know how to code a bit" through all the pieces required for a go-ipfs adapter for noms
  - was that "noms"?
- next week
  - notetaker: @em-ly
  - moderator: @lgierth

**Comment by RichardLitt** [comment]
@lgierth Can you PR the notes to this repo into `meeting_notes`, too?

**Comment by lgierth** [comment]
ooh yep!

**Comment by lgierth** [comment]
Sorry it committed it right away because I forgot changing the master-vs-PR radio button

**Comment by diasdavid** [comment]
## go-ipfs

#### Lead: @whyrusleeping
#### Notetaker: @diasdavid

#### Participants
 - @jbenet
 - @RichardLitt
 - @whyrusleeping
 - @kubuxu
 - @lgierth
 - @flyingzumwalt
 - @victorbjelkholm
 - @em-ly
 - @diasdavid

## Agenda
  - 0.4.3 review
  - js-ipfs interop milestone update
  - ipld milestone update

**You have 30 minutes for this agenda**, 5 minutes before the meeting ends, consider moving the remaining items to a github discussion thread so that all the other sprint meetings can start in time.

## Notes

- We've a "last mile" list of action items generated by Juan that need to happen before 0.4.3 release
- Some items will be ported to 0.4.4, which will be merged to master after 0.4.3 release
- js-ipfs interop
  - A new PR using the correct multicodecs is up
  - go-ipfs is going to support protocols with and without semver
  - current version of go-multistream is doing exact matching, however semver support is right there too, not the multivalidation mechanism required for the IPLD 

- We have a SECIO that works in js-ipfs, but it hasn't been tested for the go-ipfs platform
  - github issue here: https://github.com/ipfs/js-libp2p-ipfs/issues/15 (@whyrusleeping to create the scripts listed on that issue)
- multibase is implemented
- whyrusleeping has made a go-cid package
  - [ ] Create a README (@jbenet)
- go-cid will be integrated this week
  - replace the use of multihash by CIDs (because this is backwards compatible)
- Should peer-id change to use CID as well?
  - Discussion needs to happen
  - Let's piggyback on that discussion the change from /ipfs/ to /p2p/
- We are getting close to have 80% code coverage, this doesn't include the core/commands (the CLI). Next: figure out how to get code coverage from the sharness tests.
- go-ipfs Project Roadmap
- WIP Roadmap - https://docs.google.com/spreadsheets/d/1sQSkjTgft84qGs9bru6Qvegtd3dz3ROgQmLQBFMDstU/edit (not accessible, @em-ly could you check that?)
- @em-ly is working with matt to get the go-ipfs roadmap and pm set up - https://github.com/ipfs/pm/issues/153
     - current roadmap: https://github.com/ipfs/pm/issues/127
     - [ ] @em-ly yo update the github issue with what is in the google drive spreadsheet
     - @whyrusleeping has created milestones https://github.com/ipfs/go-ipfs/milestones in github that describe what needs to be done for each milestone.
- @jbenet Code Review of 0.4.3 - @whyrusleeping is reviewing it

#### @whyrusleeping wanna review these notes and PR it to the meetings notes folder + create issues (or linking to the ones already existing) for tracking purposes? :)

**Comment by RichardLitt** [comment]
@diasdavid Might be easier to make a PR, and then have @whyrusleeping review the PR.

**Comment by jbenet** [comment]
### PMing thing -- Request for people to write their mechanics down

- Please write whatever "implementations" you plan to use in this issue: https://github.com/ipfs/pm/issues/154 -- the goal is to surface them all there so we are all aware of them as we go into the mechanics.
- ATM it's missing "GH Milestones as Milestones" (@haadcode, @whyrusleeping), "GH Labels as Milestones" (@diasdavid).
- There are probably others.

**Comment by parkan** [comment]
@RichardLitt can we get an approx call time for tmw so we can plan around it?

**Comment by diasdavid** [comment]
# js-ipfs Action Items from Sprint Discussion

- [ ] pull-streams endeavour
  - [ ] Create halfClosed conn upgrade
  - [ ] Fix WebSockets issues found https://github.com/libp2p/js-libp2p-swarm/pull/67#issuecomment-242099942
  - [ ] Test libp2p against go-libp2p
    - [ ] https://github.com/ipfs/js-libp2p-ipfs/issues/15
- [ ] interface-ipfs-core endeavour
  - [ ] spec out and create tests for bitswap and run them through js-ipfs + js-ipfs-api
  - [ ] swarm tests https://github.com/ipfs/js-ipfs/issues/438 (focus on pull-streams first)
    - [ ] needs: https://github.com/ipfs/js-libp2p-ipfs/pull/16
  - [ ] https://github.com/ipfs/js-ipfs/issues/439
    - [ ] needs: https://github.com/ipfs/js-ipfs/pull/323 (@voxelot)

**Comment by jbenet** [comment]
@parkan does 1p work for you guys?

**Comment by parkan** [comment]
@jbenet yep!
